src/Swarm/Language/Typecheck.hs

```diff
@@ -774,6 +774,7 @@ inferConst c = case c of
   Scout -> [tyQ| dir -> cmd bool |]
   Whereami -> [tyQ| cmd (int * int) |]
   Waypoint -> [tyQ| text -> int -> cmd (int * (int * int)) |]
+  Structure -> [tyQ| text -> int -> cmd (unit + (int * (int * int))) |]
```

Comment:
Perhaps could remove the sum type in the return value and just indicate length of `0` instead of `inL ()`.

Reply:
Nevermind; if the return type was `cmd (int * (int * int))` instead, when length is `0`, there is nothing to populate in the right side of the tuple.  So it needs to be wrapped in the sum type.